### PR TITLE
Fix formFields value property and phone field

### PR DESCRIPTION
**Pull request by jordanranz**
_Issue #, if available:_
#6168 https://github.com/aws-amplify/amplify-js/issues/6097 https://github.com/aws-amplify/amplify-js/issues/6189 #6056 #6224

_Description of changes:_
- Fixing `formFields` value
- Fixing phone field value support and styling
- Adding support for custom dial code

Old look:
Screen Shot 2020-07-07 at 1 11 28 PM

Chrome:
Screen Shot 2020-07-07 at 12 58 17 PM

Firefox:
Screen Shot 2020-07-07 at 12 58 27 PM

Safari:
Screen Shot 2020-07-07 at 1 29 36 PM

*Note: Safari does not have the Amazon Ember font*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-country-dial-code/amplify-country-dial-code.tsx [hunk 1]
```diff
@@ -14,8 +14,21 @@ export class AmplifyCountryDialCode {
   @Prop() options: CountryCodeDialOptions = countryDialCodes;
   /** The callback, called when the input is modified by the user. */
   @Prop() handleInputChange?: (inputEvent: Event) => void;
+  /** Default selected dial code */
+  @Prop() dialCode?: string | number;
 
   render() {
-    return <amplify-select fieldId={this.fieldId} options={this.options} handleInputChange={this.handleInputChange} />;
+    let selectedDialCode = this.dialCode;
```
Comment (jordanranz) [inline packages/amplify-ui-components/src/components/amplify-country-dial-code/amplify-country-dial-code.tsx:21]: I would like to move this outside of the `render` method

File: packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.scss [hunk 2]
```diff
@@ -1,3 +1,5 @@
 :host {
   --background-color: var(--amplify-background-color);
+  -webkit-font-smoothing: antialiased;
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.scss:3]: How does this look on Safari?

Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.scss:3]: I'll add some screenshots

File: packages/amplify-ui-components/src/components/amplify-hint/amplify-hint.scss [hunk 3]
```diff
@@ -7,7 +7,7 @@
 .hint {
   color: var(--color);
   font-family: var(--font-family);
-  font-weight: 300;
+  font-weight: 400;
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-hint/amplify-hint.scss:10]: NIT: Do we want to make this a constant?

File: packages/amplify-ui-components/src/components/amplify-select/amplify-select.tsx [hunk 4]
```diff
@@ -20,12 +20,20 @@ export class AmplifySelect {
...
+  @Prop() selected?: string | number;
 
   private contructSelectOptions(opts: SelectOptionsString | SelectOptionsNumber) {
     let content = [];
-    opts.forEach((opt: SelectOptionString | SelectOptionNumber) =>
-      content.push(<option value={opt.value}>{opt.label}</option>),
-    );
+
+    opts.forEach((opt: SelectOptionString | SelectOptionNumber) => {
+      console.log(opt);
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-select/amplify-select.tsx:30]: Is this needed?

File: packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx [hunk 5]
```diff
@@ -147,6 +147,7 @@ export class AmplifySignUp {
     }
 
     try {
+      console.log(this.signUpAttributes);
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx:150]: Can we look to remove this?

File: packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx [hunk 6]
```diff
@@ -268,6 +269,24 @@ export class AmplifySignUp {
...
+            case 'username':
+              this.signUpAttributes.username = field.value;
+            case 'password':
+              this.signUpAttributes.password = field.value;
+            case 'email':
+              this.signUpAttributes.attributes.email = field.value;
+            case 'phone_number':
+              if (field.dialCode) this.phoneNumber.countryDialCodeValue = field.dialCode;
+              this.phoneNumber.phoneNumberValue = field.value;
+            // this.signUpAttributes.attributes. = field.dialCode;
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx:284]: Are these needed?

File: packages/amplify-ui-components/src/components/amplify-hint/amplify-hint.scss (hunk 3 above)
Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-hint/amplify-hint.scss:10]: yea, better yet we can make this a css variable

File: packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx [hunk 7]
```diff
@@ -160,6 +172,7 @@ export class AmplifySignIn {
     }
 
     try {
+      console.log(this.signInAttributes);
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx:175]: Is this `console.log(...)` needed?

File: packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx [hunk 8]
```diff
@@ -280,10 +284,18 @@ export class AmplifySignUp {
   setFieldValue(field: PhoneFormFieldType | FormFieldType, formAttributes: AmplifySignUpAttributes) {
     switch (field.type) {
       case 'username':
-        formAttributes.username = field.value;
+        if (field.value === undefined) {
+          formAttributes.username = '';
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx:288]: Why this change?

Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx:266]: This function handles the value prop from a custom form field to be tracked in state properly. If no value is set for `username` or `password` it will be undefined. This will revert the state value to an empty string to clean up the values passed to `Auth.signUp`.

File: packages/amplify-ui-components/src/components/amplify-auth-fields/component-field-mapping.tsx [hunk 9]
```diff
@@ -44,12 +44,12 @@ const componentFieldMapping = {
       required={ff.required}
       handleInputChange={ff.handleInputChange}
       value={ff.value}
-      inputProps={ff.inputProps}
+      inputProps={{ ...ff.inputProps, min: '0' }}
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-auth-fields/component-field-mapping.tsx:47]: Im not sure what the min 0 is for, is this there that we need to have a minimum of props coming from `inputProps` ?

Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-auth-fields/component-field-mapping.tsx:47]: @elorzafe mentioned you can use the arrows on the number input to have a negative number for the `code` field. This will prevent that field from being negative.
